framework/util/test/main.cpp

```diff
@@ -205,37 +205,45 @@ TEST_CASE("UtcString", "[datetime]")
 {
     gfxrecon::util::Log::Init(gfxrecon::util::Log::kDebugSeverity);
 
-    auto zero         = UtcString(0);
-    auto one          = UtcString(1);
-    auto two          = UtcString(2);
+    // Looking for clean output like zero: 1970-01-01T00:00:00Z
+    REQUIRE(UtcString(0) == "1970-01-01T00:00:00Z");
+    REQUIRE(UtcString(1) == "1970-01-01T00:00:01Z");
+    REQUIRE(UtcString(60 * 60) == "1970-01-01T01:00:00Z");
+    REQUIRE(UtcString(60 * 60 + 1) == "1970-01-01T01:00:01Z");
+    REQUIRE(UtcString(60 * 60 + 61) == "1970-01-01T01:01:01Z");
+    REQUIRE(UtcString(60 * 60 + 61) == "1970-01-01T01:01:01Z");
+
+    const unsigned min       = 60;
+    const unsigned hour      = min * 60;
+    const unsigned day       = hour * 24;
+    const unsigned year      = 525600 * min;
+    const unsigned leap_year = year + day;
+
+    REQUIRE(UtcString(year) == "1971-01-01T00:00:00Z");
+    REQUIRE(UtcString(year * 2) == "1972-01-01T00:00:00Z");
+    REQUIRE(UtcString(year * 8 + leap_year * 2) == "1980-01-01T00:00:00Z");
+    REQUIRE(UtcString(year * 11 + leap_year * 3 + 31 * 3 * day + 28 * day + 17 * day + 11 * hour + 28 * min + 59) ==
+            "1984-05-18T11:28:59Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7) == "2000-01-01T00:00:00Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7 + 29 * day + 31 * day * 4 + 30 * day * 2) == "2000-08-01T00:00:00Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7 + 29 * day + 31 * day * 4 + 30 * day * 2 + 30 * day) ==
+            "2000-08-31T00:00:00Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7 + 29 * day + 31 * day * 4 + 30 * day * 2 + 30 * day + 23 * hour) ==
+            "2000-08-31T23:00:00Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7 + 29 * day + 31 * day * 4 + 30 * day * 2 + 30 * day + 23 * hour +
+                      47 * min) == "2000-08-31T23:47:00Z");
+    REQUIRE(UtcString(year * 23 + leap_year * 7 + 29 * day + 31 * day * 4 + 30 * day * 2 + 30 * day + 23 * hour +
+                      47 * min + 54) == "2000-08-31T23:47:54Z");
+
     auto one_thousand = UtcString(1000);
     auto two_thousand = UtcString(2000);
-
-    // Looking for clean output like zero: 1970-00-01T00:00:00Z
-    REQUIRE(zero[18] == '0');
-    REQUIRE(zero[19] == 'Z');
-    REQUIRE(zero[4] == '-');
-    REQUIRE(zero[7] == '-');
-    REQUIRE(zero[10] == 'T');
-    REQUIRE(zero[16] == ':');
-    REQUIRE(zero[13] == ':');
-
-    REQUIRE(one[18] == '1');
-    REQUIRE(one[19] == 'Z');
-
-    REQUIRE(two[18] == '2');
-    REQUIRE(two[19] == 'Z');
-
     REQUIRE(one_thousand[19] == 'Z');
     REQUIRE(two_thousand[19] == 'Z');
-
-    REQUIRE(zero.length() == 20);
-    REQUIRE(one.length() == 20);
-    REQUIRE(two.length() == 20);
```

Comment:
These old tests were just looking at length and punctuation but not checking that the actual correct dates were being output.